modules/javafx.controls/src/test/java/test/javafx/scene/control/ComboBoxTest.java

```diff
@@ -285,6 +285,46 @@ public Node getDisplayNode() {
         assertNull(comboBox.getSelectionModel());
     }
 
+    @Test public void testNullSelectionModelDoesNotThrowNPEInSkinOnValueChange() {
+        ObservableList<String> items = FXCollections.observableArrayList("ITEM1", "ITEM2");
+
+        ListCell<String> buttonCell = new ListCell<>() {
+            @Override
+            protected void updateItem(String item, boolean empty) {
+                super.updateItem(item, empty);
+                setText(item);
+            }
+        };
+        comboBox.setButtonCell(buttonCell);
+        comboBox.setItems(items);
+        comboBox.setSelectionModel(null);
+
+        comboBox.setValue(items.get(1));
+
+        assertEquals(items.get(1), comboBox.getButtonCell().getText());
+    }
+
+    @Test public void testNullSelectionModelDoesNotThrowNPEOnValueChange() {
+        ObservableList<String> items = FXCollections.observableArrayList("ITEM1", "ITEM2");
+
+        ListCell<String> buttonCell = new ListCell<>() {
+            @Override
+            protected void updateItem(String item, boolean empty) {
+                super.updateItem(item, empty);
+                setText(item);
+            }
+        };
+
+        ComboBox<String> comboBox = new ComboBox<>();
+        comboBox.setButtonCell(buttonCell);
```

Comment:
KISS - the buttonCell or not has no effect if there is no skin :)

Reply:
oh damn, right. fixed it.